trio/_core/_run.py

```diff
@@ -1049,7 +1100,13 @@ def aborted(raise_cancel):
                 # (see test_nursery_cancel_doesnt_create_cyclic_garbage)
                 del self._pending_excs
 
-    def start_soon(self, async_fn, *args, name=None):
+    def start_soon(
+        self,
+        # TODO: TypeVarTuple
```

Comment:
I thought I saw something about mypy being able to understand `TypeVarTuple` in one of the newer updates unless I am mistaken

Reply:
It's got experimental support - definitions work, but when Mypy tries checking function calls it tends to crash with `AssertionError`s.

Reply:
https://github.com/python/mypy/labels/topic-pep-646

Reply:
If so, we should leave it for a different PR that changes it in all files though and can figure out any issues.

Reply:
You have to pass the undocumented `--enable-incomplete-feature=TypeVarTuple` + `=Unpack` to enable the incomplete code. If not, you get a warning raised for the code, `Unpack` becomes `Any`. So maybe we should indeed use them, if we can suppress the error.